src/RLECore.hs

```diff
@@ -1,26 +1,28 @@
-module RLECore where
+module RLECore (runLengthEncode, runLengthDecode) where
 
-import Data.Char
+import Data.Char (isNumber)
+
+data Encoding = Encoding Char Int
 
 runLengthEncode :: String -> String
-runLengthEncode text = go '\0' 0 [] text
+runLengthEncode = encodeRunLengths.createRunLengths
     where
-        go :: Char -> Int -> String -> String -> String
-        go ch ch_count result [] = case ch_count of
-            0 -> result
-            _ -> result ++ [ch] ++ show ch_count
-        go ch ch_count result (x:xs)
-            | x == ch       = go ch (ch_count+1) result xs
-            | ch_count == 0 = go x 1 result xs
-            | otherwise     = go x 1 (result ++ [ch] ++ show ch_count) xs
+        createRunLengths :: String -> [Encoding]
```

Comment:
Is it possible to rewrite this with https://hackage.haskell.org/package/groupBy-0.1.0.0/docs/Data-List-GroupBy.html#v:group ?

Reply:
Oops, wrong link - please refer to this one https://hackage.haskell.org/package/base-4.15.0.0/docs/Data-List.html#v:group

Reply:
Thanks this is a really useful function! Done ✅